java/opendcs/src/main/java/org/opendcs/database/JdbiTransaction.java

```diff
@@ -0,0 +1,62 @@
+package org.opendcs.database;
+
+import java.sql.Connection;
+import java.util.Optional;
+
+import org.jdbi.v3.core.Handle;
+import org.opendcs.database.api.DataTransaction;
+import org.opendcs.database.api.OpenDcsDataException;
+
+import opendcs.util.sql.WrappedConnection;
+
+public class JdbiTransaction implements DataTransaction
+{
+    final Handle jdbiHandle;
+
+    public JdbiTransaction(Handle handle)
+    {
+        this.jdbiHandle = handle;
+        if (!this.jdbiHandle.isInTransaction())
+        {
+            this.jdbiHandle.begin();
+        }
+    }
+
+    @SuppressWarnings("unchecked") // types are checked before any operations happen.
+    @Override
+    public <T> Optional<T> connection(Class<T> connectionType) throws OpenDcsDataException
+    {
+        if (Connection.class.isAssignableFrom(connectionType))
+        {
+            return (Optional<T>)Optional.of(new WrappedConnection(jdbiHandle.getConnection(), c -> {}));
+        }
+        else if (Handle.class.equals(connectionType))
+        {
+            return (Optional<T>)Optional.of(jdbiHandle);
+        }
+        else
+        {
+            return Optional.empty();
```

Comment:
What kind of connection type is this case? If the XML case, shouldn't a specific class type get sent in? using any other type seems like an error

Reply:
Nothing available, hence why the Test just passes in an integer.

I can see an argument for why maybe that it shouldn't be optional though this PR isn't the place to tweak that.

The XML database has it's own wrapper that sticks to the simple Connection for which we have an `XmlConnection` implementation to store the file location. The XML stuff is really a stop gap until we replace with SQLlite.

Though I do like Optional here. we will also have HTTP connections in the future and it's possible, though admittedly perhaps not likely, that a given implement way want to "i'd prefer there but I'll fallback on old behavior if I have to." That kind of control shouldn't be done with exceptions.

Perhaps returning `FailableResult` would be better here.

Reply:
Interesting idea, I'm not sure if Closable it entirely correct but constraining to something should be done... I will investigate a bit.

Reply:
I think adding that `extends AutoCloseable` is enough. Without any constraint it really feels like the caller has to know implementation details for what kinds of things are supported.